app/coffee/infrastructure/RateLimiter.coffee

```diff
@@ -1,15 +1,27 @@
 settings = require("settings-sharelatex")
-redis = require("redis-sharelatex")
-rclient = redis.createClient(settings.redis.web)
-redback = require("redback").use(rclient)
-
-module.exports =
-
-	addCount: (opts, callback = (opts, shouldProcess)->)->
-		ratelimit = redback.createRateLimit(opts.endpointName)
-		ratelimit.addCount opts.subjectName, opts.timeInterval, (err, callCount)->
-			shouldProcess = callCount < opts.throttle
-			callback(err, shouldProcess)
-			
+RedisWrapper = require('./RedisWrapper')
+rclient = RedisWrapper.client('ratelimiter')
+
+
+module.exports = RateLimiter =
+
+	_buildKey: (endpoint, subject) ->
+		return "RateLimiter:#{endpoint}:{#{subject}}"
+
+	addCount: (opts, callback = (err, shouldProcess)->)->
+		k = RateLimiter._buildKey(opts.endpointName, opts.subjectName)
+		multi = rclient.multi()
+		multi.incr(k)
+		multi.get(k)
```

Comment:
I guess with this system if a user tries the operation again it will reset the ttl and not expire so if a user tries to login 8 times in min 1, 3 times in min 2 and hits the rate limit they could try every minutes forever and it will still be on the rate limit. Is that how redback worked?